### PR TITLE
Fix issues with form and boxplots, issue #69

### LINKED ISSUE #69: Box plots & group boxplots issues from Artur
* [x] It seems like visualization is too large for the view. We can not fix it completely I think, but we can improve:
    - [x] Maybe try to make the ‘filter’ a bit more compact in horizontal dimension especially. May be not possible..
    - [ ] There is no obvious ‘mover’ or how is it called, to move the view. On touch-screen (mobile view) it is intuitively to do by touch and fingers. On screen is works well is one has touch screen. Otherwise not convenient.
    - [x] Make as default the ‘portrait’ view.
    - [x] Could you make a ‘visible cells’ panel horizontal, just above the main plot? It is taking too much horizontal space.. 

* [x] Similarly to starting page – we need simple icon showing what is tumor and what is stroma
* [x] Is relation to the issues above – it seems like the order of the tumors, presented on the Grouped box plots is always alphabetical. It would be easier to read if it would rather follow the selection order in the ‘filters’.. So it would show the second filter data below the first filter data. 

* [x] For Grouped box plots – could you make the colours on the group filtering panels corresponding the colour code on the graph. Now the colour in filters is dark blue everywhere.
* [x] For Grouped box plots – limit the selection to only ONE group in each of two filter panels.
* [ ] [additionally to this – but not mandatory, this is rather issue to discuss.. would it be possible to add below some option to add third (fourth) filter panel in case one needs to compare more than two groups? If complicated we just keep it limited to two at the same time]
* [x] Sort cells in the plot in the same order as the starting page splash.

### PR DESCRIPTION
**Pull request by danr**

### CODE REVIEW

**Comment by danr** [comment]
Version is up and running as 0.0.23 on http://encima.one

File: frontend/src/Form.tsx [hunk 1]
```diff
@@ -36,47 +40,27 @@ type State = Record<string, string[]> & {
...
   const state0 = {} as State
   conf.variant_values.forEach(v => {
     state0[v.column] = v.values
   })
   conf.tumor_specific_values.forEach(v => {
-    state0[v.column + ',' + v.tumor] = v.values
+    state0[specific_key(v)] = v.values
   })
-  state0.cells = []
+  state0.cells = [conf.cells.includes('CD4') ? 'CD4' : conf.cells[0]]
```
Comment (norling) [inline frontend/src/Form.tsx]: I think it'd look prettier with having 'CD4' in a `default_value` constant, but it's not important :)

File: frontend/src/BoxplotWithControls.tsx [hunk 2]
```diff
@@ -97,62 +104,86 @@ function useOptions(facet: keyof Row) {
...
+      facet.match(/cell/)
+        ? utils.sort_cells(facet_values_unsorted)
+        : utils.sort_tumors(facet_values_unsorted),
+    [facet, facet_values_unsorted]
+  )
+
+  const [store] = useStore({visible: facet_values})
 
-  const [visible_facets, facet_boxes] = ui.useCheckboxes(facet_values, all_selected, utils.pretty)
+  // React.useLayoutEffect(() => {
```
Comment (norling) [inline frontend/src/BoxplotWithControls.tsx]: I would remove the commented code.

File: frontend/src/BoxplotWithControls.tsx [hunk 3]
```diff
@@ -97,62 +104,86 @@ function useOptions(facet: keyof Row) {
...
+  )
+
+  const value = Object.fromEntries(store.get().visible.map(v => [v, true]))
+
+  const [delayed_value, enqueue_delayed_value] = ui.useDelayed(400, value)
+
+  const loading = !utils.equal(value, delayed_value)
+
+  React.useEffect(() => {
+    enqueue_delayed_value(value)
```
Comment (norling) [inline frontend/src/BoxplotWithControls.tsx:135]: why does the value need to be delayed? It's probably correct - but an explanation would help!

**Pull request merged**